pkg/graveler/retention/expired_commits.go

```diff
@@ -12,37 +14,68 @@ type GarbageCollectionCommits struct {
 	active  []graveler.CommitID
 }
 
+type CommitNode struct {
+	CommitID     graveler.CommitID
+	CreationDate time.Time
+	ParentsIDs   []graveler.CommitID
+	Version      graveler.CommitVersion
+}
+
+func NewCommitNode(commitID graveler.CommitID, creationDate time.Time, parentsIDs []graveler.CommitID, version graveler.CommitVersion) *CommitNode {
+	return &CommitNode{
+		CommitID:     commitID,
+		CreationDate: creationDate,
+		ParentsIDs:   parentsIDs,
+		Version:      version,
+	}
+}
+
+var ErrCommitNotFound = errors.New("commit not found")
+
 // GetGarbageCollectionCommits returns the sets of expired and active commits, according to the repository's garbage collection rules.
 // See https://github.com/treeverse/lakeFS/issues/1932 for more details.
 // Upon completion, the given startingPointIterator is closed.
 func GetGarbageCollectionCommits(ctx context.Context, startingPointIterator *GCStartingPointIterator, commitGetter *RepositoryCommitGetter, rules *graveler.GarbageCollectionRules, previouslyExpired []graveler.CommitID) (*GarbageCollectionCommits, error) {
 	// From each starting point in the given startingPointIterator, it iterates through its main ancestry.
 	// All commits reached are added to the active set, until and including the first commit performed before the start of the retention period.
 	// All further commits in the ancestry are added to the expired set. The iteration stops upon reaching a commit which exists in the previouslyExpired set, or the DAG root.
-	now := time.Now()
 	processed := make(map[graveler.CommitID]time.Time)
 	previouslyExpiredMap := make(map[graveler.CommitID]bool)
 	for _, commitID := range previouslyExpired {
 		previouslyExpiredMap[commitID] = true
 	}
 	activeMap := make(map[graveler.CommitID]struct{})
 	expiredMap := make(map[graveler.CommitID]struct{})
+
+	commitsIterator, err := commitGetter.ListCommits(ctx)
+	if err != nil {
+		return nil, err
+	}
+	commitsMap := make(map[graveler.CommitID]*CommitNode)
```

Comment:
Do we need a pointer here?  Asking because we were counting memory AFAIR, and I guess the pointer here adds at least 16 bytes memory (8 to hold the pointer, another 8 for another object header).